### PR TITLE
Merge FlowableEmitter.BackpressureMode into BackpressureStrategy

### PR DESCRIPTION
**Pull request by adipascu**
This is the implementation of #4727

### CODE REVIEW

File: src/main/java/io/reactivex/Flowable.java [hunk 1]
```diff
@@ -9735,6 +9735,12 @@ public final Completable ignoreElements() {
         return RxJavaPlugins.onAssembly(new FlowableOnBackpressureLatest<T>(this));
     }
 
+    @BackpressureSupport(BackpressureKind.UNBOUNDED_IN)
```
Comment (akarnokd) [inline src/main/java/io/reactivex/Flowable.java]: This is the default behavior of most Flowable sources. Unnecessary.

File: src/main/java/io/reactivex/Flowable.java [hunk 2]
```diff
@@ -9735,6 +9735,12 @@ public final Completable ignoreElements() {
...
+    @SchedulerSupport(SchedulerSupport.NONE)
+    /*package*/ Flowable<T>  onBackpressureError() {
+        return RxJavaPlugins.onAssembly(new FlowableOnBackpressureError<T>(this));
```
Comment (akarnokd) [inline src/main/java/io/reactivex/Flowable.java]: Should be inlined in `Observable.toFlowable()`.

File: src/main/java/io/reactivex/internal/operators/flowable/FlowableCreate.java [hunk 3]
```diff
@@ -61,10 +61,13 @@ public void subscribeActual(Subscriber<? super T> t) {
             emitter = new LatestAsyncEmitter<T>(t);
             break;
         }
-        default: {
+        case BUFFER: {
```
Comment (akarnokd) [inline src/main/java/io/reactivex/internal/operators/flowable/FlowableCreate.java]: Why changing this?

File: src/test/java/io/reactivex/internal/operators/observable/ObservableToXTest.java [hunk 4]
```diff
@@ -48,4 +49,31 @@ public void toFlowableLatest() {
         ts
         .assertResult(5);
     }
+
+    @Test
+    public void toFlowableError1() {
+        TestSubscriber<Integer> ts = Observable.range(1, 5)
+        .toFlowable(BackpressureStrategy.ERROR)
+        .test(1)
+        .assertError(MissingBackpressureException.class);
```
Comment (akarnokd) [inline src/test/java/io/reactivex/internal/operators/observable/ObservableToXTest.java]: Why not `assertFailure(MissingBackpressureException.class, 1)` to make sure `1` got through.

File: src/main/java/io/reactivex/Observable.java [hunk 5]
```diff
@@ -11707,12 +11707,18 @@ public final void subscribe(Observer<? super T> observer) {
...
-            return o.onBackpressureBuffer();
+            case DROP:
+                return o.onBackpressureDrop();
+            case LATEST:
+                return o.onBackpressureLatest();
+            case NONE:
+                return o;
+            case ERROR:
+                return o.onBackpressureError();
+            case BUFFER:
```
Comment (akarnokd) [inline src/main/java/io/reactivex/Observable.java]: Please leave these as `default:` because otherwise the default case can't be coverage-tested.

**Review by akarnokd — CHANGES_REQUESTED** [review]

**Comment by adipascu** [comment]
I have implemented the changes from the code comments.
I'll keep in mind to rely on code coverage in the future.
Should I reply to the review comments directly ?

**Review by akarnokd — APPROVED** [review]

**Comment by akarnokd** [comment]
No need in this case. Looks good to me.

**Pull request merged**